packages/prompts/src/autocomplete.ts

```diff
@@ -0,0 +1,287 @@
+	const hint = (option.hint ?? '').toLowerCase();
+	const value = String(option.value).toLowerCase();
+	const term = searchText.toLowerCase();
+
+	return label.includes(term) || hint.includes(term) || value.includes(term);
+}
+
+function getSelectedOptions<T>(values: T[], options: Option<T>[]): Option<T>[] {
+	const results: Option<T>[] = [];
+
+	for (const option of options) {
+		if (values.includes(option.value)) {
+			results.push(option);
+		}
+	}
+
+	return results;
+}
+
+export interface AutocompleteOptions<Value> extends CommonOptions {
+	/**
+	 * The message to display to the user.
+	 */
+	message: string;
+	/**
+	 * Available options for the autocomplete prompt.
+	 */
+	options: Option<Value>[];
+	/**
+	 * The initial selected value.
+	 */
+	initialValue?: Value;
+	/**
+	 * Maximum number of items to display at once.
+	 */
+	maxItems?: number;
+	/**
+	 * Placeholder text to display when no input is provided.
+	 */
+	placeholder?: string;
+}
+
+export const autocomplete = <Value>(opts: AutocompleteOptions<Value>) => {
+	const prompt = new AutocompletePrompt({
+		options: opts.options,
+		placeholder: opts.placeholder,
+		initialValue: opts.initialValue ? [opts.initialValue] : undefined,
+		filter: (search, opt) => {
+			return getFilteredOption(search, opt);
+		},
+		input: opts.input,
+		output: opts.output,
+		render() {
+			// Title and message display
+			const title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+			const valueAsString = String(this.value ?? '');
+
+			// Handle different states
+			switch (this.state) {
+				case 'submit': {
+					// Show selected value
+					const selected = getSelectedOptions(this.selectedValues, this.options);
+					const label = selected.length > 0 ? selected.map(getLabel).join(', ') : '';
+					return `${title}${color.gray(S_BAR)}  ${color.dim(label)}`;
+				}
+
+				case 'cancel': {
+					return `${title}${color.gray(S_BAR)}  ${color.strikethrough(color.dim(this.value ?? ''))}`;
+				}
+
+				default: {
+					// Display cursor position - show plain text in navigation mode
+					const searchText = this.isNavigating ? color.dim(valueAsString) : this.valueWithCursor;
+
+					// Show match count if filtered
+					const matches =
+						this.filteredOptions.length !== this.options.length
+							? color.dim(
+									` (${this.filteredOptions.length} match${this.filteredOptions.length === 1 ? '' : 'es'})`
+								)
+							: '';
+
+					// Render options with selection
+					const displayOptions =
+						this.filteredOptions.length === 0
+							? []
+							: limitOptions({
+									cursor: this.cursor,
+									options: this.filteredOptions,
+									style: (option, active) => {
+										const label = getLabel(option);
+										const hint = option.hint ? color.dim(` (${option.hint})`) : '';
+
+										return active
+											? `${color.green(S_RADIO_ACTIVE)} ${label}${hint}`
+											: `${color.dim(S_RADIO_INACTIVE)} ${color.dim(label)}${hint}`;
```

Comment:
@MacFJA suggested this shouldn't have a `hint` since it isn't the active option

probably is right (the comment was lost in the refactor)